src/app.ts

```diff
@@ -35,19 +35,25 @@ declare global {
   namespace Express {
     interface Request {
       generatedSessionId?: string;
+      cri?: {
+        id?: string;
+        redirectUrl?: string;
+      };
+      redirectURL?: URL;
```

Comment:
I don't think we need the redirectUrl in the cri obj but `criHelper.buildCredentialIssuerRedirectURL` attaches a redirectUrl to the request

Reply:
Hmm I see - that's pretty weird, it looks like we call `buildCredentialIssuerRedirectURL`, attach the redirect to the request, and then call `redirectToAuthorize` which picks it off and redirects??

Can we just generate the redirect as a string, and then redirect using that string? I don't see any need to put it on the request object. (Same for the `cri` property, which also seems to be placed onto the request and immediately picked off again).

I know this is nothing to do with you, but this has helped highlight the weirdness!

Reply:
I've ended up just removing this helper and placing the logic directly into ipv middleware because it was being used in just one place and could be simplified quite a bit